sql/core/src/main/scala/org/apache/spark/sql/execution/UnsafeRowSerializer.scala

```diff
@@ -143,7 +143,7 @@ private class UnsafeRowSerializerInstance(numFields: Int) extends SerializerInst
       override def readKey[T: ClassTag](): T = {
         // We skipped serialization of the key in writeKey(), so just return a dummy value since
         // this is going to be discarded anyways.
-        null.asInstanceOf[T]
+        (-1).asInstanceOf[T]
```

Comment:
Actually this looks really weird now and doesn't really match the comment. Would you mind changing it back?